File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Projects/getnumberofpages.rb
module Asposetasksjava
  module GetNumberOfPages
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        presentation_format = Rjb::import('com.aspose.tasks.PresentationFormat')
        timescale = Rjb::import('com.aspose.tasks.Timescale')

        # Get number of pages
        puts "Number of Pages = " + project.getPageCount(presentation_format.ResourceUsage, timescale.Days).to_s

        # Get number of pages (Months)
        puts "Number of Pages = " + project.getPageCount(presentation_format.ResourceUsage, timescale.Months).to_s

        # Get number of pages (ThirdsOfMonths)
        puts "Number of Pages = " + project.getPageCount(presentation_format.ResourceUsage, timescale.ThirdsOfMonths).to_s
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithProjects/GetNumberOfPages.php
<?php
namespace Aspose\Tasks\WorkingWithProjects;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\PresentationFormat as PresentationFormat;
use com\aspose\tasks\Timescale as Timescale;

class GetNumberOfPages{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $presentation_format = new PresentationFormat();
        $timescale = new Timescale();

        # Get number of pages
        print "Number of Pages = " . (string)$project->getPageCount($presentation_format->ResourceUsage, $timescale->Days).PHP_EOL;//.to_s

        # Get number of pages (Months)
        print "Number of Pages = " . (string)$project->getPageCount($presentation_format->ResourceUsage, $timescale->Months).PHP_EOL;//.to_s

        # Get number of pages (ThirdsOfMonths)
        print "Number of Pages = " . (string)$project->getPageCount($presentation_format->ResourceUsage, $timescale->ThirdsOfMonths).PHP_EOL;//.to_s

    }
}
?>